var/spack/repos/builtin/packages/cxx/package.py

```diff
@@ -14,28 +14,36 @@ class Cxx(Package):
     homepage = "https://isocpp.org/std/the-standard"
     virtual = True
 
-    def test(self):
+    def test_c(self):
+        """Compile and run 'Hello World'"""
         test_source = self.test_suite.current_test_data_dir
 
-        for test in os.listdir(test_source):
-            filepath = os.path.join(test_source, test)
-            exe_name = "%s.exe" % test
-
-            cxx_exe = os.environ["CXX"]
-
-            # standard options
-            # Hack to get compiler attributes
-            # TODO: remove this when compilers are dependencies
-            c_name = clang if self.spec.satisfies("llvm+clang") else self.name
-            c_spec = spack.spec.CompilerSpec(c_name, self.spec.version)
-            c_cls = spack.compilers.class_for_compiler_name(c_name)
-            compiler = c_cls(c_spec, None, None, ["fakecc", "fakecxx"])
+        cxx_exe = os.environ["CXX"]
+        cxx_exe = which(join_path(self.prefix.bin, cxx_exe))
+        if cxx_exe is None:
+            raise SkipTest(f"{os.environ['CXX']} not found in {self.version}")
 
-            cxx_opts = [compiler.cxx11_flag] if "c++11" in test else []
-
-            cxx_opts += ["-o", exe_name, filepath]
-            compiled = self.run_test(cxx_exe, options=cxx_opts, installed=True)
-
-            if compiled:
-                expected = ["Hello world", "YES!"]
-                self.run_test(exe_name, expected=expected)
+        for test in os.listdir(test_source):
+            with test_part(self, f"test_cxx_{test}", f"Test {test}"):
+                filepath = os.path.join(test_source, test)
+                exe_name = f"{test}.exe"
+                # standard options
+                # Hack to get compiler attributes
+                # TODO: remove this when compilers are dependencies
+                c_name = clang if self.spec.satisfies("llvm+clang") else self.name
+                c_spec = spack.spec.CompilerSpec(c_name, self.spec.version)
+                c_cls = spack.compilers.class_for_compiler_name(c_name)
+                compiler = c_cls(c_spec, None, None, ["fakecc", "fakecxx"])
+                cxx_opts = [compiler.cxx11_flag] if "c++11" in test else []
+                cxx_opts += ["-o", exe_name, filepath]
+                compiled = cxx_exe(*cxx_opts)
```

Comment:
Calling `Executable` does not return the same value as calling `self.run_test()` so we don't usually do that UNLESS passing `output` and `error` arguments (see line 46).